PostHog/App Life Cycle/PostHogAppLifeCycleIntegration.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  PostHogAppLifeCycleIntegration.swift
+//  PostHog
+//
+//  Created by Ioannis Josephides on 19/02/2025.
+//
+
+import Foundation
+
+/**
+ Add capability to capture application lifecycle events.
+
+ This integration:
+ - captures an `App Installed` event on the first launch of the app
+ - captures an `App Updated` event on any subsequent launch with a different version
+ - captures an `App Opened` event when the app is opened (including the first launch)
+ - captures an `App Backgrounded` event when the app moves to the background
+ */
+final class PostHogAppLifeCycleIntegration {
+    private static var integrationInstalledLock = NSLock()
+    private static var integrationInstalled = false
+
+    private weak var postHog: PostHogSDK?
+
+    private var appFromBackground = false
+
+    private var didBecomeActiveToken: RegistrationToken?
+    private var didEnterBackgroundToken: RegistrationToken?
+    private var didFinishLaunchingToken: RegistrationToken?
+    private var didCaptureAppInstalled = false
+
+    init?(_ posthog: PostHogSDK) {
+        let wasInstalled = PostHogAppLifeCycleIntegration.integrationInstalledLock.withLock {
+            if PostHogAppLifeCycleIntegration.integrationInstalled {
+                hedgeLog("App life cycle integration already installed to another PostHogSDK instance.")
+                return true
+            }
+            PostHogAppLifeCycleIntegration.integrationInstalled = true
+            return false
+        }
+
+        guard !wasInstalled else { return nil }
+
+        postHog = posthog
+
+        // needs to happen on next runloop cause we are still in SDK init sequence here
+        DispatchQueue.main.async {
+            self.captureAppInstallOrUpdated()
+        }
```

Comment:
I think to avoid this we can do this:
https://github.com/PostHog/posthog-android/blob/41ac8e9dbf989348823052c8c0f14c9c0d1fa2fd/posthog/src/main/java/com/posthog/PostHog.kt#L125-L131
we create the instances, and "install" them once the SDK is operational at the end of the setup method